### DISCUSSION TITLE
When I use Dokodemo-door in router, can I still use domain routing rules?

### DISCUSSION THREAD

**Discussion by sgon00** [v2fly/v2ray-core#2889] [Q&A]
In routing rules, people always have something like:

```json
"routing": {
        "domainStrategy": "IpIfNonMatch",
        "rules": [
            {
                "type": "field",
                "outboundTag": "direct",
                "domain": [
                    "geosite:cn"
                ]
            },

```

But, I am very confused how it works when I use Dokodemo-door protocol in router?
Normally I will use `iptables` command to forward all traffics to Dokodemo-door port. But iptables only cares about IPs only. When the traffics forward to Dokodemo-door port, all traffics are IPs only from my point of view. `geosite:cn` domain will not take any effects. Am I wrong? How can I make domain routing rules make sense and work in a router setup? 

Thanks a lot.

**Comment by om2mo** [comment]
iptables转发的是第四层TCP/UDP数据包，因此它包含了IP地址。
```
ex:
iptables  -p udp -dport 53 to 1234
dig qq.com
```
iptables会将查询qq.com的 dns udp 包发往1234，dns udp = src ip/port, dst ip/port, domain name.

进入dokodemo port可以是域名和IP，这取决于你的需求，所以geosite策略是有效的。
